client/components/ShareButton.js

```diff
@@ -23,16 +23,18 @@ export default () => {
   const shareData = {url: window.location.href};
   const classes = useStyles(navigator.canShare && navigator.canShare(shareData))();
 
-  const share = (shareData) => {
-    navigator.share(shareData).catch((e) => {
-      if (e.name !== "AbortError") {
+  const share = async (shareData) => {
+    try {
+      await navigator.share(shareData);
+    } catch (e) {
+      if (e.name !== 'AbortError') {
         // Failure wasn't due to the user deciding to cancel the share.
         setState({error: e.toString(), showSnackbar: true});
       }
-    });
+    }
   };
 
-  const closeSnackbar = () => setState({showSnackbar: false});
+  const closeSnackbar = () => setState({error: state.error, showSnackbar: false});
```

Comment:
Not propagating the error message was causing a brief flash while the snackbar was being closed.

Reply:
Ack, will make sure to check for that when I refactor it :-)